llvm/lib/CodeGen/LiveRangeShrink.cpp

```diff
@@ -125,14 +135,19 @@ bool LiveRangeShrink::runOnMachineFunction(MachineFunction &MF) {
     if (MBB.empty())
       continue;
     bool SawStore = false;
+    bool IsEHPad = MBB.isEHPad();
+    MachineInstr *EHBarrier = nullptr;
     BuildInstOrderMap(MBB.begin(), IOM);
     UseMap.clear();
 
     for (MachineBasicBlock::iterator Next = MBB.begin(); Next != MBB.end();) {
       MachineInstr &MI = *Next;
       ++Next;
+
       if (MI.isPHI() || MI.isDebugOrPseudoInstr())
         continue;
+      if (IsEHPad && !EHBarrier && MI.isEHLabel())
```

Comment:
Should this be using SkipPHIsLabelsAndDebug?

Reply:
Well it could have been using it if we did not want to check for labels but in my fix i use the first ehlabel to determine the start of the landingpad.